export.js

```diff
@@ -0,0 +1,161 @@
+function grabExcelValues(rows, artifact, objTemplate, customFieldRange) {
+    if (rows === null) {
+        getRows(artifact, objTemplate, customFieldRange);
+    }
+    else {
+        return Excel.run(function (context) {
+            let sheet = context.workbook.worksheets.getItem(toSheetName(artifact));
+            let inputRange = columnRanges[artifact] + rows;
+            let customRange = (customFieldRange[0] + 3) + ":" + (customFieldRange[1] + rows);
+            let inputValues = sheet.getRange(inputRange);
+            let customFields = sheet.getRange(customRange);
+            inputValues.load();
+            customFields.load();
+            return context.sync()
+                .then(function () {
+                    customFieldObjArr = customFields.values.map(customFieldObjCreate);
+                    ajaxExport(inputValues.values, artifact, objTemplate, customFieldObjArr);
+                })
+        });
+    }
+}
+
+function getRows(artifact, objTemplate, customFieldRange) {
+    return Excel.run(function (context) {
+        let sheet = context.workbook.worksheets.getItem(toSheetName(artifact));
+        let sheetRange = sheet.getUsedRange();
+        sheetRange.load();
+        return context.sync()
+            .then(function () {
+                grabExcelValues(sheetRange.values.length, artifact, objTemplate, customFieldRange);
+            })
+            .catch(function (error) {
+                console.log(error);
+            })
+    });
+}
+
+function ajaxExport(valueArray, artifact, objTemplate, customFieldObjArr) {
```

Comment:
could probably happen all at once, currently only different stages because the main fields were working before I had started on the custom fields. will look into for next week